File: app/src/main/java/com/drdisagree/iconify/xposed/EntryList.java
package com.drdisagree.iconify.xposed;

/* Modified from AOSPMods
 * https://github.com/siavash79/AOSPMods/blob/canary/app/src/main/java/sh/siava/AOSPMods/modpacks/ModPacks.java
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see [http://www.gnu.org/licenses/].
 */

import static com.drdisagree.iconify.common.Const.PIXEL_LAUNCHER_PACKAGE;
import static com.drdisagree.iconify.common.Const.SYSTEMUI_PACKAGE;

import android.os.Build;

import com.drdisagree.iconify.xposed.modules.BackgroundChip;
import com.drdisagree.iconify.xposed.modules.BatteryStyleManager;
import com.drdisagree.iconify.xposed.modules.DepthWallpaper;
import com.drdisagree.iconify.xposed.modules.HeaderClock;
import com.drdisagree.iconify.xposed.modules.HeaderImage;
import com.drdisagree.iconify.xposed.modules.IconUpdater;
import com.drdisagree.iconify.xposed.modules.LockscreenClock;
import com.drdisagree.iconify.xposed.modules.Miscellaneous;
import com.drdisagree.iconify.xposed.modules.QSTransparency;
import com.drdisagree.iconify.xposed.modules.QuickSettings;
import com.drdisagree.iconify.xposed.modules.themes.QSBlackThemeA13;
import com.drdisagree.iconify.xposed.modules.themes.QSBlackThemeA14;
import com.drdisagree.iconify.xposed.modules.themes.QSFluidThemeA13;
import com.drdisagree.iconify.xposed.modules.themes.QSFluidThemeA14;
import com.drdisagree.iconify.xposed.modules.themes.QSLightThemeA12;
import com.drdisagree.iconify.xposed.modules.themes.QSLightThemeA13;
import com.drdisagree.iconify.xposed.modules.themes.QSLightThemeA14;
import com.drdisagree.iconify.xposed.modules.utils.SettingsLibUtils;
import com.drdisagree.iconify.xposed.utils.HookCheck;

import java.util.ArrayList;

public class EntryList {

    public static ArrayList<Class<? extends ModPack>> getEntries(String packageName) {
        ArrayList<Class<? extends ModPack>> modPacks = new ArrayList<>();

        modPacks.add(SettingsLibUtils.class);
        modPacks.add(HookCheck.class);

        switch (packageName) {
            case SYSTEMUI_PACKAGE -> {
                if (!HookEntry.isChildProcess) {
                    modPacks.add(BackgroundChip.class);
                    modPacks.add(HeaderClock.class);
                    modPacks.add(HeaderImage.class);
                    modPacks.add(DepthWallpaper.class);
                    modPacks.add(LockscreenClock.class);
                    modPacks.add(Miscellaneous.class);
                    modPacks.add(QSTransparency.class);
                    modPacks.add(QuickSettings.class);
                    modPacks.add(BatteryStyleManager.class);

                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) { // Android 14+
                        modPacks.add(QSFluidThemeA14.class);
                        modPacks.add(QSBlackThemeA14.class);
                        modPacks.add(QSLightThemeA14.class);
                    } else if (Build.VERSION.SDK_INT == Build.VERSION_CODES.TIRAMISU) { // Android 13
                        modPacks.add(QSFluidThemeA13.class);
                        modPacks.add(QSBlackThemeA13.class);
                        modPacks.add(QSLightThemeA13.class);
                    } else { // Android 12 and below
                        modPacks.add(QSFluidThemeA13.class);
                        modPacks.add(QSBlackThemeA13.class);
                        modPacks.add(QSLightThemeA12.class);
                    }
                }
            }
            case PIXEL_LAUNCHER_PACKAGE -> modPacks.add(IconUpdater.class);
        }

        return modPacks;
    }
}

File: app/src/main/java/com/drdisagree/iconify/xposed/EntryList.kt
package com.drdisagree.iconify.xposed

import com.drdisagree.iconify.data.common.Const.FRAMEWORK_PACKAGE
import com.drdisagree.iconify.data.common.Const.SYSTEMUI_PACKAGE
import com.drdisagree.iconify.xposed.modules.extras.ActivityLauncherUtils
import com.drdisagree.iconify.xposed.modules.extras.ExpandableViews
import com.drdisagree.iconify.xposed.modules.extras.GraphicsColorKt
import com.drdisagree.iconify.xposed.modules.extras.LaunchableViews
import com.drdisagree.iconify.xposed.modules.extras.MyConstraintSet
import com.drdisagree.iconify.xposed.modules.extras.SettingsLibUtils
import com.drdisagree.iconify.xposed.modules.extras.callbacks.ConfigurationCallback
import com.drdisagree.iconify.xposed.modules.extras.callbacks.ControllersProvider
import com.drdisagree.iconify.xposed.modules.extras.callbacks.DozeCallback
import com.drdisagree.iconify.xposed.modules.extras.callbacks.HeadsUpCallback
import com.drdisagree.iconify.xposed.modules.extras.callbacks.KeyguardShowingCallback
import com.drdisagree.iconify.xposed.modules.extras.callbacks.QsShowingCallback
import com.drdisagree.iconify.xposed.modules.extras.callbacks.ThemeChangeCallback
import com.drdisagree.iconify.xposed.modules.lockscreen.AlbumArt
import com.drdisagree.iconify.xposed.modules.lockscreen.DepthWallpaper
import com.drdisagree.iconify.xposed.modules.lockscreen.Lockscreen
import com.drdisagree.iconify.xposed.modules.lockscreen.LockscreenClock
import com.drdisagree.iconify.xposed.modules.lockscreen.LockscreenWeather
import com.drdisagree.iconify.xposed.modules.lockscreen.LockscreenWidgets
import com.drdisagree.iconify.xposed.modules.misc.Miscellaneous
import com.drdisagree.iconify.xposed.modules.quicksettings.AppIconInNotification
import com.drdisagree.iconify.xposed.modules.quicksettings.ColorizeNotificationView
import com.drdisagree.iconify.xposed.modules.quicksettings.HeaderClock
import com.drdisagree.iconify.xposed.modules.quicksettings.HeaderImage
import com.drdisagree.iconify.xposed.modules.quicksettings.HeadsUpBlur
import com.drdisagree.iconify.xposed.modules.quicksettings.QSGrid
import com.drdisagree.iconify.xposed.modules.quicksettings.QSTheme
import com.drdisagree.iconify.xposed.modules.quicksettings.QSTransparency
import com.drdisagree.iconify.xposed.modules.quicksettings.QuickSettings
import com.drdisagree.iconify.xposed.modules.statusbar.AppIconsInStatusbar
import com.drdisagree.iconify.xposed.modules.statusbar.BatteryStyleManager
import com.drdisagree.iconify.xposed.modules.statusbar.ClockChip
import com.drdisagree.iconify.xposed.modules.statusbar.DualStatusbar
import com.drdisagree.iconify.xposed.modules.statusbar.OnGoingActionChip
import com.drdisagree.iconify.xposed.modules.statusbar.StatusbarLogo
import com.drdisagree.iconify.xposed.modules.statusbar.StatusbarMisc
import com.drdisagree.iconify.xposed.modules.statusbar.SwapSignalNetworkType
import com.drdisagree.iconify.xposed.modules.statusbar.SwapWiFiCellular
import com.drdisagree.iconify.xposed.modules.volume.VolumePanel
import com.drdisagree.iconify.xposed.utils.HookCheck

object EntryList {

    private val topPriorityCommonModPacks: List<Class<out ModPack>> = listOf(
        SettingsLibUtils::class.java,
        HookCheck::class.java
    )

    private val systemUIModPacks: List<Class<out ModPack>> = listOf(
        /* Top priority */
        GraphicsColorKt::class.java,
        MyConstraintSet::class.java,
        LaunchableViews::class.java,
        ExpandableViews::class.java,
        ActivityLauncherUtils::class.java,
        ControllersProvider::class.java,
        ThemeChangeCallback::class.java,
        HeadsUpCallback::class.java,
        QsShowingCallback::class.java,
        KeyguardShowingCallback::class.java,
        DozeCallback::class.java,
        ConfigurationCallback::class.java,
        /* Not so top priority :P */
        BatteryStyleManager::class.java,
        QSGrid::class.java,
        ClockChip::class.java,
        HeaderImage::class.java,
        HeaderClock::class.java,
        Lockscreen::class.java,
        LockscreenWidgets::class.java,
        LockscreenWeather::class.java,
        AlbumArt::class.java,
        Miscellaneous::class.java,
        QSTransparency::class.java,
        QuickSettings::class.java,
        AppIconsInStatusbar::class.java,
        SwapWiFiCellular::class.java,
        SwapSignalNetworkType::class.java,
        DualStatusbar::class.java,
        StatusbarMisc::class.java,
        VolumePanel::class.java,
        ColorizeNotificationView::class.java,
        AppIconInNotification::class.java,
        HeadsUpBlur::class.java,
        OnGoingActionChip::class.java,
        StatusbarLogo::class.java,
        DepthWallpaper::class.java,
        LockscreenClock::class.java,
        LockscreenWeather::class.java,
        LockscreenWidgets::class.java,
        QSTheme::class.java
    )

    fun getEntries(packageName: String): ArrayList<Class<out ModPack>> {
        val modPacks = ArrayList<Class<out ModPack>>()

        modPacks.addAll(topPriorityCommonModPacks)

        when (packageName) {
            FRAMEWORK_PACKAGE -> {}

            SYSTEMUI_PACKAGE -> {
                if (!HookEntry.isChildProcess) {
                    modPacks.addAll(systemUIModPacks)
                }
            }
        }

        return modPacks
    }
}
